### PR TITLE
feat!: Add UnsupportedError to unnest_to_explode transform

### PR DESCRIPTION
**Pull request by VaggelisD**
Follow up to #3999

Refactor `generator` arg out of `unnest_to_explode()`

### CODE REVIEW

File: sqlglot/transforms.py [hunk 1]
```diff
@@ -308,11 +312,7 @@ def _unnest_zip_exprs(
     ) -> t.List[exp.Expression]:
         if has_multi_expr:
             if not unnest_using_arrays_zip:
-                if generator:
-                    generator.unsupported(
-                        f"Multiple expressions in UNNEST are not supported in "
-                        f"{generator.dialect.__module__.split('.')[-1].upper()}"
-                    )
+                raise UnsupportedError("Cannot transpile UNNEST with multiple arrays to Hive")
```
Comment (georgesittas) [inline sqlglot/transforms.py:315]: This is dialect-agnostic:
```suggestion
                raise UnsupportedError("Cannot transpile UNNEST with multiple arrays")
```

File: sqlglot/transforms.py [hunk 2]
```diff
@@ -27,9 +29,12 @@ def preprocess(
...
 
-        expression = transforms[0](expression)
-        for transform in transforms[1:]:
-            expression = transform(expression)
+        try:
+            expression = transforms[0](expression)
+            for transform in transforms[1:]:
+                expression = transform(expression)
+        except UnsupportedError as unsupported_error:
+            self.unsupported(str(unsupported_error))
```
Comment (georgesittas) [inline sqlglot/transforms.py:37]: Did you test that this actually logs the correct warning? I don't remember if `str(...)` is the correct way to get the message.

**Review by georgesittas — APPROVED** [review]

File: sqlglot/transforms.py (hunk 2 above)
Comment (VaggelisD) [reply sqlglot/transforms.py:37]: Yep, here's an example:

```
>>> sqlglot.parse_one('SELECT * FROM example_table CROSS JOIN UNNEST(split(type, ";"), scores)', read="starrocks").sql("hive")
Cannot transpile UNNEST with multiple arrays
"SELECT * FROM example_table CROSS JOIN EXPLODE(SPLIT(type, CONCAT('\\\\Q', ';')), scores, unnest(unnest))"
```

Comment (georgesittas) [reply sqlglot/transforms.py:37]: Cool thanks for double checking

**Pull request merged**